SAMtemplates/load_balancer/main.yaml

```diff
@@ -0,0 +1,146 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::Serverless-2016-10-31
+Description: |
+  Main template for Load Balancer with all resources included, supporting mutual TLS
+
+Parameters:
+  StackName:
+    Type: String
+
+  EnableMutualTLS:
+    Type: String
+    AllowedValues:
+      - true
+      - false
+
+  TruststoreVersion:
+    Type: String
+
+  TruststoreFile:
+    Type: String
+
+  VPC:
+    Type: AWS::EC2::VPC::Id
+
+  SubnetA:
+    Type: AWS::EC2::Subnet::Id
+
+  SubnetB:
+    Type: AWS::EC2::Subnet::Id
+
+  SubnetC:
+    Type: AWS::EC2::Subnet::Id
+
+  VersionNumber:
+    Type: String
+
+  CommitId:
+    Type: String
+
+  HostedZoneName:
+    Type: String
+
+Conditions:
+  CreateTruststore: !Equals [!Ref EnableMutualTLS, true]
+
+Resources:
+  # Generate a new ACM certificate
+  GenerateCertificate:
+    Type: AWS::CertificateManager::Certificate
+    Properties:
+      ValidationMethod: DNS
+      DomainName:
+        Fn::Join:
+          - "."
+          - - !Ref StackName
+            - !ImportValue eps-route53-resources:EPS-domain
+      DomainValidationOptions:
+        - DomainName:
+            Fn::Join:
+              - "."
+              - - !Ref StackName
+                - !ImportValue eps-route53-resources:EPS-domain
+          HostedZoneId: !ImportValue eps-route53-resources:EPS-ZoneID
+
+  # Create a new Application Load Balancer
+  LoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      LoadBalancerAttributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: 60
+      Scheme: internet-facing
+      SecurityGroups:
+        - !ImportValue vpc-resources:VpcDefaultSecurityGroup
+      Subnets:
+        - !Ref SubnetA
```

Comment:
Theres no need to pass these in from the top level template as they arnt referencing values from other sub stacks and are instead referencing outputs from account resources. So just directly import them here.

Also account resources as an output for the subnets as a list (PublicSubnets), it would be cleaner to just use that one output than the 3 individual ones.

Reply:
The parameters have been updated and improved.